runtime/alloc/collect.c

```diff
@@ -92,11 +92,11 @@ static void migrate_string_buffer(stringbuffer** bufferPtr) {
     string *newContents;
     if (shouldPromote) {
       newBuffer = koreAllocOld(sizeof(stringbuffer));
-      memcpy(newBuffer, buffer, sizeof(stringbuffer));
+      memcpy(newBuffer, buffer, sizeof(uint64_t)); // contents is written below
```

Comment:
This probably gets optimized, but shouldn't this just be a simple assignment now that it only copies the block header? In any case, this should use `sizeof(block)` rather than hardcode the header size as `uint64_t`.